File: Bubble_Sort/bubble_sort.cpp
#include <iostream>

void bubbleSort(int arr[], int n) {
    bool flag = false;
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                std::swap(arr[j], arr[j+1]);
                flag = true;
            }
        }
        if(!flag) {
            break;
        }
    }
}

// A utility function to print an array of size n
void printArray(int l[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        std::cout << l[i] << " ";
    std::cout << std::endl;
}

// Driver code to test above
int main() {
    
    int arr[] = { 12, 11, 13, 5, 6 };
    int n = sizeof(arr) / sizeof(arr[0]);

    bubbleSort(arr, n);
    printArray(arr, n);
    
    return 0;
}

File: Bubble_Sort/BubbleSort.py
# Implementation of Bubble Sort
def bubble_sort(arr: list) -> None:
    for i in range(len(arr)):
        swapped = False
        for j in range(len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped: break
     
        
# Driver code to test above
if __name__ == "__main__":
    nums = [6, 1, 5, 9, 9, 5, 3, 8, 0]
    bubble_sort(nums)
    print(nums)
